### PR TITLE
Add 'naxsi_attack_action' variable

### PR DESCRIPTION
**Pull request by giovanirinaldi**
This variable, named 'naxsi_attack_action', exports two types of
 information regarding the treatment of the current request.

First, it indicates if naxsi is currently under learning mode,
by prefixing the value of this variable with 'LEARNING-'. Second,
it shows if the request has been blocked or not, using the values
'BLOCK' or 'PASS', thus resulting in the following four possible
compound values:

- $PASS: the request has been accepted
- $BLOCK: the request was blocked
- $LEARNING-PASS: naxsi is in learning mode and the request
                  has been accepted
- $LEARNING-BLOCK: the request would be blocked but, due to being
                   in learning mode, it was not.

### CODE REVIEW

**Comment by marcelomd** [comment]
:+1

**Comment by roberthbelter** [comment]
LGTM 👍

File: naxsi_src/naxsi_skeleton.c [hunk 1]
```diff
@@ -1383,6 +1393,86 @@ ngx_http_naxsi_attack_family_variable(ngx_http_request_t *r, ngx_http_variable_v
...
+            break;
+    }
+
+    if (sz == 0) {
+        v->not_found = 1;
+        return NGX_OK;
+    }
+
+    str = (u_char *)ngx_pcalloc(r->pool, sz);
+    p = str;
```
Comment (FelipeEcker) [inline naxsi_src/naxsi_skeleton.c]: If heap allocate fail here? Whats happen?
The pointer *p can be NULL and memcpy() can show one unexpected error.

We can test the allocate fail ? (str != NULL)

Comment (giovanirinaldi) [inline naxsi_src/naxsi_skeleton.c]: Fixed, thank you!

**Comment by FelipeEcker** [comment]
:

**Pull request merged**